### PR TITLE
Issue/3174 update cef

### PR DESCRIPTION
**Pull request by ylvaselling**
This PR updates CEF to the newest version.

Testing on multiple platform is encouraged :)

### CODE REVIEW

File: modules/webbrowser/src/cefhost.cpp [hunk 1]
```diff
@@ -44,6 +44,9 @@ CefHost::CefHost([[maybe_unused]] const std::string& helperLocation) {
     LDEBUG("Initializing CEF...");
 
     CefSettings settings;
+    const std::string root = helperLocation.substr(0, helperLocation.rfind("\\"));
+    const std::string cefcache = std::format("{}\\{}", root, "cefcache");
+    CefString(&settings.root_cache_path).FromString(cefcache);
```
Comment (alexanderbock) [inline modules/webbrowser/src/cefhost.cpp:49]:
```suggestion
    const std::filesystem::path root =
        std::filesystem::path(helperLocation).parent_path();
    const std::filesystem::path cefcache = std::format("{}/cefcache", root);
    CefString(&settings.root_cache_path).FromString(cefcache.string());
```

**Comment by GPayne** [comment]
I tested this on linux. Unfortunately it crashed with this console output:
```
(D) CefHost              Initializing CEF...
(D) CefHost              Remote WebBrowser debugging available on http://localhost:8088
[0819/220115.562502:WARNING:alloy_main_delegate.cc(559)] Alloy bootstrap is deprecated and will be removed in ~M127. See https://github.com/chromiumembedded/cef/issues/3685
[0819/220115.566821:ERROR:icu_util.cc(223)] Invalid file descriptor to ICU data received.
Trace/breakpoint trap (core dumped)
```

**Review by alexanderbock — APPROVED** [review]
Worked for me on Windows

**Comment by wtbgagoa** [comment]
After the CEF update I am no longer able to build OpenSpace. I get the following error :
```
/home/alfonso/openspace-git/src/OpenSpace/build/modules/webbrowser/ext/cef/cef_binary_127.3.5+g114ea2a+chromium-127.0.6533.120_linux64/include/base/cef_ref_counted.h:487:42: error: expected ‘)’ before ‘,’ token
  487 |   explicit RefCountedData(std::in_place_t - , Args&&... args)
                                                                                     
```
I tried with gcc 14 and gcc 13. I also tried clang, which reports a different error at the same line of code:
```
/home/alfonso/openspace-git/src/OpenSpace/build/modules/webbrowser/ext/cef/cef_binary_127.3.5+g114ea2a+chromium-127.0.6533.120_linux64/include/base/cef_ref_counted.h:487:32: error: no type named 'in_place_t' in namespace 'std'
  487 |   explicit RefCountedData(std::in_place_t, Args&&... args)

```
@GPayne: which compiler did you use in your test ?

**Comment by GPayne** [comment]
@wtbgagoa: I am also getting CEF build errors, but they are related to a later commit for CEF GPU acceleration. I created [issue#3391](https://github.com/OpenSpace/OpenSpace/issues/3391) for this.
If you go back a few days in the github history to commit 6e6f425, it should build for you. To answer your compiler question, I am building with gcc 13.

**Comment by wtbgagoa** [comment]
> @wtbgagoa: I am also getting CEF build errors, but they are related to a later commit for CEF GPU acceleration. I created [issue#3391](https://github.com/OpenSpace/OpenSpace/issues/3391) for this. If you go back a few days in the github history to commit [6e6f425](https://github.com/OpenSpace/OpenSpace/commit/6e6f425a0a655d14bc5bfe4b734831194d355cf5), it should build for you. To answer your compiler question, I am building with gcc 13.

Commit [6e6f425](https://github.com/OpenSpace/OpenSpace/commit/6e6f425a0a655d14bc5bfe4b734831194d355cf5) doesn't build for me (it gives me the same error). I need to go back to [d0b3cd6](https://github.com/OpenSpace/OpenSpace/commit/d0b3cd6e8903ae55d1d918b35cb67231a85296f1) (right before CEF upgrade) to be able to build with either compiler.

**Pull request merged**